application/controllers/admin/Database.php

```diff
@@ -364,12 +364,15 @@ protected function actionUpdateSurveyLocaleSettingsGeneralSettings($surveyId)
             'admin' => $request->getPost('admin'),
             'adminemail' => $request->getPost('adminemail'),
             'bounce_email' => $request->getPost('bounce_email'),
-            'gsid' => $request->getPost('gsid'),
             'format' => $request->getPost('format'),
             'owner_id' => $request->getPost('owner_id'),
             'template' => $request->getPost('template')
         ];
-
+        if (SurveysGroups::model()->permission(App()->user->getId())->findByPk(intval($request->getPost('gsid')))) {
```

Comment:
You will probably don't like my comment, still, I feel need to.
On this line we are intending a business rule: Does the user has permission for the survey group?
Shouldn't we enclose that on a method somewhere so we can better reuse it in the future and enhance it if needed?

Reply:
>  On this line we are intending a "business" rule / "access" rule: Does the user has permission for the survey group?

It's in model, permission scope … done for this, like Survey->permission
Like `SurveyGroup::getSurveyGroupsList`

>  Shouldn't we enclose that on a method somewhere so we can better reuse it in the future and enhance it if needed?

It's in scope : it's **clearly** sufficient. 

We can just move it to PermissionTrait and Interface. **But** when i want to add it : after a long , too long discussion i remove it and close the PR. You can not ask now to add it again … 

And here !: it's a **FIX** , it's not a rewrite … you want i check some previous PR where need

Reply:
Ese : before rewrite something clean in model : need to move all admin/database.php elwewhere …